File: l.c
#include<stdio.h>
#include<stdlib.h>
#define min(a,b) (a<b?a:b)
int main()
{
    /*int D[4][4]={
        {0,5,999,10},
        {999,0,3,999},
        {999,999,0,1},
        {999,999,999,0}
    };*/
    int i,j,n;
    printf("enter the number of vertices:");
    scanf("%d",&n);
    int D[n][n];
    printf("enter the adjacency weight matrix:");
    for(i=0;i<n;i++)
    {
        printf("vertex %d:\n",i+1);
        for(j=0;j<n;j++)
        {
            scanf("%d",&D[i][j]);
        }
    }
    /*int D[6][6]={
        {0,999,999,999,-1,999},
        {1,0,999,2,999,999},
        {999,2,0,999,999,-8},
        {-4,999,999,0,3,999},
        {999,7,999,999,0,999},
        {999,5,10,999,999,0}
    };*/

    /*int D[3][3]={
        {0,5,999},
        {999,0,2},
        {999,999,0} 
    };*/
    int k;
    int x;
    for(k=1;k<n;k++)
    {
        for(i=0;i<n;i++)
        {
            for(j=0;j<n;j++)
            {
                x=(D[i][k]+D[k][j]);
                D[i][j]=min(D[i][j],x);
            }
        }
    }
    for(i=0;i<n;i++)
    {
        for(j=0;j<n;j++)
        {
            if(D[i][j]>900)
            {
                printf("INF     ");
            }
            else
            {    
            printf("%d     ",D[i][j]);
            }
        }
        printf("\n");
    }
    return 0;
}

File: l.py
class Graph:
	def __init__(self,v,e):
		self.v=v
		self.e=e
		self.edges=[]
	def addEdge(self):
		
		print("enter the edges with their weight in the format (src dest weight)")
	
		for i in range(self.e):
			str=input()
			str=str.split()
			self.edges.append([int(str[0]),int(str[1]),int(str[2])])
	def bellmanFord(self,src):
		dist=[[float("inf") for i in range(self.v)]for j in range(self.v)]
		dist[src]=[0 for j in range(self.v)]
		for k in range(1,self.v):
			for v in range(self.v):
				dist[v][k]=dist[v][k-1]
				for u,v1,w in self.edges:
					if(v1==v):
						if dist[u][k-1]!=float("inf") and (dist[v][k]>(dist[u][k-1]+w)):
							dist[v][k]=dist[u][k-1]+w

		for u,v,w in self.edges:
			if dist[v][self.v-1]>dist[u][self.v-1]+w:
				print("Negative cycle:")
				return
		print("table:")
		print(dist)





def main():
	v=int(input("enter the total number of vertice:"))
	e=int(input("enter the total number of edges:"))
	g=Graph(v,e)
	g.addEdge()
	print("enter the source vertex:")
	src=int(input())
	g.bellmanFord(src)

main()
